core/domain/playthrough_issue_registry_test.py

```diff
@@ -18,14 +18,53 @@
 
 from __future__ import annotations
 
+from core import utils
 from core.domain import playthrough_issue_registry
 from core.tests import test_utils
+from extensions.issues.CyclicStateTransitions import CyclicStateTransitions
+from extensions.issues.EarlyQuit import EarlyQuit
+from extensions.issues.MultipleIncorrectSubmissions import (
+    MultipleIncorrectSubmissions)
 
 
 class IssueRegistryUnitTests(test_utils.GenericTestBase):
     """Test for the issue registry."""
 
+    def setUp(self):
+        super(IssueRegistryUnitTests, self).setUp()
+        self.issues_dict = {
+                'EarlyQuit':
+                EarlyQuit.EarlyQuit,
+                'CyclicStateTransitions':
+                CyclicStateTransitions.CyclicStateTransitions,
+                'MultipleIncorrectSubmissions':
+                MultipleIncorrectSubmissions.MultipleIncorrectSubmissions
+                }
+        self.invalid_issue_type = 'InvalidIssueType'
+        self.Registry = playthrough_issue_registry.Registry()
+
     def test_issue_registry(self):
         """Do some sanity checks on the issue registry."""
         self.assertEqual(
-            len(playthrough_issue_registry.Registry.get_all_issues()), 3)
+            len(self.Registry.get_all_issues()), 3)
+
+    def test_issue_registry_types(self):
+        """Do some issue type checks on the issue registry."""
+
+        def validate(issue_type):
```

Comment:
Don't add logic in tests; try to keep them as simple and declarative as possible. I would suggest dropping this function and inlining the relevant parts in your test.

Also perhaps use separate tests for the valid and invalid cases. Make sure your test names are descriptive. See guidance here: https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests

Reply:
I totally agree . Let me rewrite this test function asap :)